src/components/menu/menu.test.js

```diff
@@ -0,0 +1,1260 @@
+      segmentTitle()
+        .should("have.text", "segment title")
+        .and("be.visible")
+        .and("have.css", "color", "rgba(0, 0, 0, 0.65)");
+    });
+
+    it("should verify default menu clickToOpen element does not open on hover", () => {
+      CypressMountWithProviders(<MenuComponent />);
+
+      menuComponent(positionOfElement("fourth")).trigger("mouseover", {
+        force: true,
+      });
+      submenuItem(positionOfElement("fourth"))
+        .should("have.length", 0)
+        .and("not.exist");
+    });
+
+    it("should verify default menu clickToOpen element opens on click", () => {
+      CypressMountWithProviders(<MenuComponent />);
+
+      menuComponent(positionOfElement("fifth")).click();
+      submenuItem(positionOfElement("fifth")).should("have.length", 3);
+      innerMenu(positionOfElement("second"), span)
+        .should("have.attr", "data-component", "link")
+        .and("be.visible");
+      innerMenu(positionOfElement("third"), div)
+        .should("have.attr", "data-component", "menu-segment-title")
+        .and("be.visible");
+      innerMenu(positionOfElement("fourth"), span)
+        .should("have.attr", "data-component", "link")
+        .and("be.visible");
+    });
+
+    it.each(["Enter", "Space", "downarrow", "uparrow"])(
+      "should verify default menu clickToOpen element opens using %s key",
+      (key) => {
+        CypressMountWithProviders(<MenuComponent />);
+
+        menuComponent(positionOfElement("fifth")).trigger(
+          "keydown",
+          keyCode(key)
+        );
+        submenuItem(positionOfElement("fifth")).should("have.length", 3);
+        innerMenu(positionOfElement("second"), span)
+          .should("have.attr", "data-component", "link")
+          .and("be.visible");
+        innerMenu(positionOfElement("third"), div)
+          .should("have.attr", "data-component", "menu-segment-title")
+          .and("be.visible");
+        innerMenu(positionOfElement("fourth"), span)
+          .should("have.attr", "data-component", "link")
+          .and("be.visible");
+      }
+    );
+
+    it("should verify the Search component is focusable by using the downarrow key", () => {
+      CypressMountWithProviders(<MenuComponentSearch />);
+
+      pressTABKey(1);
+      cy.wait(50);
+      cy.focused().trigger("keydown", keyCode("Enter"));
+      cy.wait(50);
+      cy.focused().trigger("keydown", keyCode("downarrow"));
+      searchDefaultInput().should("have.focus");
+    });
+
+    it("should verify the Search component is focusable by using the uparrow key", () => {
+      CypressMountWithProviders(<MenuComponentSearch />);
+
+      pressTABKey(1);
+      cy.wait(50);
+      cy.focused().trigger("keydown", keyCode("Enter"));
+      cy.wait(50);
+      pressTABKey(2);
+      cy.wait(50);
+      cy.focused().trigger("keydown", keyCode("uparrow"));
+      searchDefaultInput().should("have.focus");
+    });
```

Comment:
Can we combine this test case with the above test case using it.each["downarrow", "uparrow"]?